curated_transformers/pipe.py

```diff
@@ -207,6 +225,14 @@ def add_listener(
         self, listener: "TransformerListener", component_name: str
     ) -> None:
         """Add a listener for a downstream component. Usually internals."""
+        if (
+            not isinstance(listener, LastTransformerLayerListener)
```

Comment:
I think this check should be in the listener, since there can be other custom listeners that only use the last layer.

Reply:
Added `needs_all_layer_outputs` to `TransformerListener`.